cephfs/admin/subvolume.go

```diff
@@ -423,3 +427,129 @@ func (fsa *FSAdmin) UnprotectSubVolumeSnapshot(volume, group, subvolume, name st
+	Exclude    bool     `json:"exclude,omitempty"`
+	Reset      bool     `json:"reset,omitempty"`
+	Release    bool     `json:"release,omitempty"`
+	Query      bool     `json:"query,omitempty"`
+	All        bool     `json:"all,omitempty"`
+	Cancel     bool     `json:"cancel,omitempty"`
+}
+
+// SubVolumeQuiesceOptions are used to specify optional, non-identifying, values
+// to be used when quiescing a subvolume.
+type SubVolumeQuiesceOptions struct {
+	Timeout    MyFloat
+	Expiration MyFloat
+	AwaitFor   MyFloat
+	Await      bool
+	IfVersion  int
+	Include    bool
+	Exclude    bool
+	Reset      bool
+	Release    bool
+	Query      bool
+	All        bool
+	Cancel     bool
+}
+
+// toFields is used to convert the SubVolumeQuiesceOptions to the internal
+// SubVolumeQuiesceFields type.
+func (o *SubVolumeQuiesceOptions) toFields(volume, group string, subvolumes []string, SetId string) *SubVolumeQuiesceFields {
+	return &SubVolumeQuiesceFields{
+		Prefix:     "fs subvolume quiesce",
+		VolName:    volume,
+		GroupName:  group,
+		Members:    subvolumes,
+		SetId:      SetId,
+		Timeout:    o.Timeout,
+		Expiration: o.Expiration,
+		AwaitFor:   o.AwaitFor,
+		Await:      o.Await,
+		IfVersion:  o.IfVersion,
+		Include:    o.Include,
+		Exclude:    o.Exclude,
+		Reset:      o.Reset,
+		Release:    o.Release,
+		Query:      o.Query,
+		All:        o.All,
+		Cancel:     o.Cancel,
+	}
+}
+
+// QuiesceInfo reports various informational values about a quiesced subvolume.
+// This is returned as sets object array in the json.
+type QuiesceInfo struct {
+	DbVersion int     `json:"db_version"`
+	AgeRef    float64 `json:"age_ref"`
+	State     struct {
+		Name string  `json:"name"`
+		Age  float64 `json:"age"`
+	} `json:"state"`
+	Timeout    float64 `json:"timeout"`
+	Expiration float64 `json:"expiration"`
+	Members    map[string]struct {
+		Excluded bool `json:"excluded"`
+		State    struct {
+			Name string  `json:"name"`
+			Age  float64 `json:"age"`
+		} `json:"state"`
+	} `json:"members"`
+}
+
+// SubVolumeQuiesceInfo reports various informational values about a quiesced subvolume.
+type SubVolumeQuiesceInfo struct {
+	Epoch     int                    `json:"epoch"`
+	DbVersion int                    `json:"db_version"`
+	Sets      map[string]QuiesceInfo `json:"sets"`
+}
+
+// parseSubVolumeQuiesceInfo is used to parse the response from the quiesce command. It returns a SubVolumeQuiesceInfo object.
+func parseSubVolumeQuiesceInfo(res response) (*SubVolumeQuiesceInfo, error) {
+	var info SubVolumeQuiesceInfo
+	if err := res.NoStatus().Unmarshal(&info).End(); err != nil {
+		return nil, err
+	}
+	return &info, nil
+}
+
+// SubVolumeQuiesce will quiesce the specified subvolume.
+// Quiescing a subvolume will prevent new writes to the subvolume.
+// Similar To:
+//
+//	ceph fs subvolume quiesce <volume>
+func (fsa *FSAdmin) SubVolumeQuiesce(volume, group string, subvolumes []string, SetId string, o *SubVolumeQuiesceOptions) (*SubVolumeQuiesceInfo, error) {
```

Comment:
SubvolumeQuiesce support multiple group and multiple subvolumes but here it looks like only one group is supported, can we please check on that?

Reply:
i tested the feature with cephcsi, it's working as expected (at least for now :D ) we need to pass `NoGroup`, and all the subvolume []string contain `group/subvolumename` array.